### PR TITLE
bpo-36150: Fix possible assertion failures due to _ctypes.c's PyCData_reduce()

### PR DESCRIPTION
**Pull request by ZackerySpytz**
(This is a "skip news" PR.)

https://bugs.python.org/issue36150

### CODE REVIEW

**Comment by tiran** [comment]
The extra checks are not necessary. O and N handle NULL correctly.

File: Modules/_ctypes/_ctypes.c [hunk 1]
```diff
@@ -2731,11 +2731,16 @@ PyCData_reduce(PyObject *myself, PyObject *args)
...
-    return Py_BuildValue("O(O(NN))",
-                         _unpickle,
-                         Py_TYPE(myself),
-                         PyObject_GetAttrString(myself, "__dict__"),
-                         PyBytes_FromStringAndSize(self->b_ptr, self->b_size));
+    PyObject *dict = PyObject_GetAttrString(myself, "__dict__");
+    if (dict == NULL) {
+        return NULL;
+    }
+    PyObject *bytes = PyBytes_FromStringAndSize(self->b_ptr, self->b_size);
```
Comment (serhiy-storchaka) [inline Modules/_ctypes/_ctypes.c]: It is enough to check for error only `dict`. `PyBytes_FromStringAndSize()` can be passed directly to `Py_BuildValue()`.

Comment (serhiy-storchaka) [reply Modules/_ctypes/_ctypes.c]: You could even use `y#` instead of `PyBytes_FromStringAndSize()`.

Comment (ZackerySpytz) [reply Modules/_ctypes/_ctypes.c]: Thank you, Serhiy. I've updated the PR.

**Review by serhiy-storchaka — APPROVED** [review]

**Pull request merged**